### ISSUE TITLE
Add lua-cjson as dependency

### ISSUE THREAD

**Issue by jpy-luke** [menu-hanke/m2#5]
For any practical use of this program and the upcoming live development of the mela 2 simulator, the JSON I/O is needed. Thus this dependency can hardly be considered optional.

* Add the openresty/lua-cjson repo as submodule.
* Introduce the build for this module as dynamic and static for Linux, and as dynamic for Windows dll. The cross-compiled Windows dll possibility does not seem to be in any obvious way documented and as such not supported for the module. Try to determine if this is even a viable solution to keep up with.

**Comment by vfprintf** [comment]
> For any practical use of this program and the upcoming live development of the mela 2 simulator, the JSON I/O is needed.

I don't think this is true. We use JSON currently because it's handy for prototyping, but most likely the final format will instead be some binary format (something like RSD but more flexible).

**Comment by jpy-luke** [comment]
Thank you. Good to know for posterity about the build.

> I don't think this is true. We use JSON currently because it's handy for prototyping, but most likely the final format will instead be some binary format (something like RSD but more flexible). If anything, it should be a dependency of LAME instead.

I disagree somewhat here. m2 is the entry point of the application and LAME (or other) appears as a library-like implementation. m2 should thus be in charge of the I/O functionality. Furthermore and regardless of which component is in charge of the I/O, any prototyping of a full applications does require some I/O facility and as it stands now, it is the lua-cjson as a direct dependency of m2. Whether it should be JSON or something else should be a separate discussion.

Based on the discussion in #4 however, let's just close this then if this work is not going to see use as you suggest.
